### PR TITLE
Add One-Cold transform

### PR DESCRIPTION
**Pull request by ChristianWieck**
This PR:

- Adds a `OneCold` transform.

### CODE REVIEW

**Review by schmidma — DISMISSED** [review]
I'd suggest to reuse the implementation of the one-hot encoding and just take the inverse of that... What do you think?

**Comment by ChristianWieck** [comment]
I decided to implement the `OneCold` transform by copying most of the `OneHot` transform. Just applying a `OneHot` transform and inverting it unfortunately isn't that easy. I could implement a shared base that is used by both transforms, but tbh I doubt it's worth it for two transforms.

File: src/flowcean/transforms/one_cold.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        """
+        if len(self.feature_category_mapping) == 0:
+            raise NoCategoriesError
+        for (
+            feature,
+            category_mappings,
+        ) in self.feature_category_mapping.items():
+            data = data.with_columns(
+                [
+                    pl.col(feature).ne(value).cast(pl.Int64).alias(name)
```
Comment (newLabAspect) [inline src/flowcean/transforms/one_cold.py:95]: As the polars expressions here and in lines 100-107 are quite hard to grasp, I am wondering whether it's worth to add some inline comments or extend the doc-string?

File: src/flowcean/transforms/one_cold.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        self,
+        feature_categories: dict[str, list[Any]],
+        *,
+        check_for_missing_categories: bool = False,
+    ) -> None:
+        """Initializes the One-Hot transform.
+
+        Args:
+            feature_categories: Dictionary of features and a list of
+                categorical values to encode for each.
```
Comment (newLabAspect) [inline src/flowcean/transforms/one_cold.py:57]: Provide an example for the dictionary? (e.g., with respect to the example above)

File: src/flowcean/transforms/one_cold.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            features: Name of the features for which the one hot transformation
+                will determine the categories.
+            check_for_missing_categories: If set to true, a check is performed
+                to see if all values belong to a category. If an unknown value
+                is found which does not belong to any category, a
+                NoMatchingCategoryError is thrown. To perform this check, the
+                dataframe must be materialised, resulting in a potential
+                performance decrease. Therefore it defaults to false.
+        """
+        # Derive categories from the data frame
```
Comment (newLabAspect) [inline src/flowcean/transforms/one_cold.py:135]: If we decide not to put inline comments, we should probably remove this one as well.

**Review by newLabAspect — COMMENTED** [review]
The implementation looks fine to me. Just had a hard time to understand what's going on. Thus, some suggestions on how to improve this for users.

File: src/flowcean/transforms/one_cold.py (hunk 3 above)
Comment (ChristianWieck) [reply src/flowcean/transforms/one_cold.py:162]: Do you mean the flag or the comment? I think it's still relevant for the user to have the choice - they don't need to understand how it works under the hood as long as it does what they want :D

File: src/flowcean/transforms/one_cold.py (hunk 2 above)
Comment (ChristianWieck) [reply src/flowcean/transforms/one_cold.py:75]: Added a code example to the "main-docstring".

File: src/flowcean/transforms/one_cold.py [hunk 4]
```diff
@@ -39,6 +39,14 @@ class OneCold(Transform):
      1        | 1         | 0         | 1
      1        | 0         | 1         | 1
      1        | 1         | 1         | 0
+
+     In the default configuration missing categories are ignored.
+     There respective entries will all be one. If you however want to enforce
```
Comment (newLabAspect) [inline src/flowcean/transforms/one_cold.py:44]: There -> Their

**Review by newLabAspect — COMMENTED** [review]
Found a minor typo. Maybe we can fix that directly ;)

File: src/flowcean/transforms/one_cold.py (hunk 3 above)
Comment (newLabAspect) [reply src/flowcean/transforms/one_cold.py:162]: I just meant the comment ;)

File: src/flowcean/transforms/one_cold.py (hunk 4 above)
Comment (ChristianWieck) [reply src/flowcean/transforms/one_cold.py]: Should be fixed.

**Review by newLabAspect — APPROVED** [review]
lgtm now

**Comment by ChristianWieck** [comment]
@schmidma what do you think?

**Pull request merged**